server/src/services/projectService.ts

```diff
@@ -597,6 +598,64 @@ const setScheduleData = async (schedulePayload: SetSchedulePayload) => {
   return statusPayload;
 }
 
+const editScheduleData = async (payload: SetSchedulePayload) => {
+  try {
+    const scheduleData: ScheduleDataMongoResponse = await ScheduleDataSchema.findOne({"projectID": payload.projectID});
+    const keyData: ScheduleKeys | null = scheduleData.scheduleKeys.get(payload.locationID) ?? null;
+    
+    if (keyData) {
+      const parts = keyData.key.split(" ");
+      const date = parts.slice(0, 3).join(' ');
+      const time = parts.slice(3).join(' ');
+      const validDate = date.replace(/(\d+)(st|nd|rd|th),/, '$1,');
```

Comment:
i like this regex validation!

Reply:
Thank you :) it took a lot of trial and error :')